substrate/primitives/statement-store/src/runtime_api.rs

```diff
@@ -179,7 +219,7 @@ pub trait StatementStore {
 	}
 
 	/// Remove a statement from the store by hash.
-	fn remove(&mut self, hash: &Hash) {
+	fn remove(&mut self, hash: PassPointerAndRead<&Hash, 32>) {
```

Comment:
nit: maybe `std::mem::size_of::<Hash>()` instead of hard-coded 32?

Reply:
This was deliberately done to make it explicit how much data is actually being passed.

Normally, yes, I agree, you'd use `size_of` to make sure that if the size of the type changes then the code doesn't need to be updated, but notice here that *we never want this to change*! If the size of `Hash` changes then this would be a backwards incompatible change, and any historical block which uses this host function would explode.

So making this explicit is not only there to make it immediately obvious how much data is being passed, but also as an additional safeguard to protect against unintended breakage.